nucypher/network/nodes.py

```diff
@@ -421,50 +483,60 @@ def remember_node(self, node, force_verification_check=False, record_fleet_state
                 # This node is already known.  We can safely return.
                 return False
 
+        self.known_nodes[node.checksum_address] = node
+
+        if self.save_metadata:
+            self.node_storage.store_node_metadata(node=node)
+
         try:
             stranger_certificate = node.certificate
         except AttributeError:
-            # Whoops, we got an Alice, Bob, or someone...
-            raise self.NotATeacher(f"{node.__class__.__name__} does not have a certificate and cannot be remembered.")
+            # Probably a sprout.
+            try:
```

Comment:
I wonder if it's easier to do it not at try/except but by overriding `__getattr__`, so that we grow it once we first access the certificate (or maybe something else, too).

Or is it too implicit?

Reply:
Yeah, this occurred to me.  I thought it was too dramatic and perhaps a premature optimization.  I am trying to be somewhat reserved in this PR.  But jeez man, if you thought of it independently, that's pretty solid evidence that it's a good idea.

I think if we do it, let's do it all the way down in the bytestring-splitter project.